### ISSUE TITLE
Not removing pokemon from database when it expires

### ISSUE THREAD

**Issue by dfrankes** [AHAAAAAAA/PokemonGo-Map#1749]
## Expected Behavior

It should remove the pokemon from the database once it expired

## Current Behavior

a expired pokemon is still in the database after it expired

## Steps to Reproduce (for bugs)

1. Start the script
2. Look at the database file
3. Wait untill the pokemon despawns
4. Check database if its still there

## Context

Trying to get a high acurate updated .db file
## Your Environment

* Version used:  0eef97f
* Environment name and version (e.g. Python 2.7): Python 2.7.12
* Operating System and version (desktop or mobile): Windows 7 x64

**Comment by AppliedEllipsis** [comment]
I do not think this should be the case, just do a select for non-expired.  For heat map analysis and other data mining, keeping the archived data is a good thing.

**Comment by mcclurej06** [comment]
This could be implemented with as a config option.

**Comment by dfrankes** [comment]
@AppliedEllipsis How would i filter out the non expired ones? if it doesn't give me a good timestamp.

**Comment by AppliedEllipsis** [comment]
dfrankes, are you just concerned about the timestamp name?  expire-time is a ISO8601 datetime object.  SQLite only has a limited datatypes... Technically they could store epoch in int form.  But I'm seeing it in a very usable form still.  eg. "2016-07-23 19:52:25.492000"
What's wrong with something as simple as:
`SELECT * FROM `pokemon` where `disappear_time` > datetime('now')`

**Comment by Dutzen** [comment]
Don't do this! If we keep the Pokemon in the database with it's expired / despawn time  and possible spawn time, we can analyze possible spawn pattern!

**Comment by arthurkay** [comment]
-1

Historical data analysis should be a future feature (heat mapping).

**Comment by FabianSperrle** [comment]
It would be nice to have a parameter that does that! I'm currently running this on Heroku, and there is a limit of 10,000 rows in the database.

**Comment by Talamar1** [comment]
I agree that keeping history should be a feature, but the default should be to delete the records as unwary users will fill databases and crash systems.  Maybe a separate thread that does database pruning once per (??? setable parameter defaults to 1 sec?), no need to flood the sqlite/mysql system with queries and deletes multiple times per second.

**Comment by Vivicus** [comment]
I'm actually asking for an enhancement to keep stale pokemon for determining patterns- see  #1610 for what I'm talking about.  Removing old pokemon from the database would remove a lot of potential value in terms of statistical correlation, unless this was a mode *specifically* to eschew old pokemon to save on server space (For instance, in the case of high-volume, small step searches).

**Comment by ewold** [comment]
I agree that this should be an argument or option flag to clean up the old data after x time. This can be 2 hours or a week depending on what the user wants. I'm afraid the querying would get slower if the database has 100.000's of rows in it. Especially if the user is still using the sqlite backend. This can be an extra thread that checks every hour.
